src/index.js

```diff
@@ -1,51 +1,132 @@
 const path = require('path');
+// eslint-disable-next-line node/no-unpublished-require
+const getNodeProperty = require('../src/utils.js');
 
 /**
- * Write the test metadata expressions either into the body of the existing beforeEach, or
- * wrapped in a new beforeEach which is inserted just above the given babelPath
- * @param {object} state - Babel state
- * @param {object} babelPath - Babel path
- * @param {object} t  - Babel types
- * @param {boolean} hasBeforeEach  - if true, write expressions into existing beforeEach,
- *   otherwise write a new beforeEach
+ * Checks if the call expression matches a test setup call pattern.
+ * @param {object} Babel node object
+ * @param {object} t Babel types
+ * @param {string} hooksIdentifier - "hooks" passed into module's function arg
+ * @returns {Boolean} - if the call expression has a setup call pattern,
+ *   e.g. setupApplication(hooks), then return true
  */
-function writeTestMetadataExpressions(state, babelPath, t, hasBeforeEach) {
-  const testMetadataVarDeclaration = getTestMetadataDeclaration(state, t);
-  const testMetadataAssignment = getTestMetadataAssignment(state, t);
+function isSetupCall({ node }, t, hooksIdentifier) {
+  return (
+    t.isCallExpression(node.expression) &&
+    t.isIdentifier(node.expression.arguments[0]) &&
+    node.expression.arguments[0].name === hooksIdentifier
+  );
+}
 
-  if (hasBeforeEach) {
-    const functionBlock = babelPath.get('arguments')[0];
-    const functionBlockBody = functionBlock.get('body');
-    const functionBlockBodyStatementsArray = functionBlockBody.get('body');
-    let existingMetadataDeclaration;
+/**
+ * Gets the last setup call expression node path.
+ * @param {array} callsArray Babel array of call expression node paths representing calls within a function block
+ * @param {object} t Babel types
+ * @param {string} hooksIdentifier - "hooks" passed into module's function arg
+ * @returns {object} a Babel node path that is the last setup call expression, e.g. setupApplication(hooks)
+ */
+function getLastSetupCall(callsArray, t, hooksIdentifier) {
+  if (callsArray.length === 1 && isSetupCall(callsArray[0], t, hooksIdentifier))
+    return callsArray[0];
 
-    if (functionBlockBodyStatementsArray.length > 0) {
-      existingMetadataDeclaration = functionBlockBodyStatementsArray.find(
-        hasMetadataDeclaration
-      );
+  for (let i = 0; i < callsArray.length; i++) {
+    if (
+      callsArray[i + 1] &&
```

Comment:
I think I'd have to see more of the surrounding stuff to get more context. But if it was something like:
`module('Integration | Component | bar', function(hooks) { 
  setupRenderingTest(hooks);
  setupTwo(hooks);
  setupThree(hooks);

  hooks.beforeEach(function(){...});
  hooks.afterEach();
  hooks.beforeEach();
`
The plugin, traversing top down on each call, would check each of the `anything(hooks)` calls and store the last one, setupThree(hooks).

Then it would check for any beforeEach calls and store the first beforeEach. That's the beforeEach it would add our statements to.